enhancements/update/targeted-update-edge-blocking.md

```diff
@@ -0,0 +1,180 @@
+
+The graph-builder's graph-data scraper should learn about [the new 1.1.0 schema](#enhanced-graph-data-schema-for-blocking-edges), and record any `clusters` properties in edge metadata.
+
+When a request is received, the submitted [`channel` query parameter][cincinnati-for-openshift-request] limits the set of remaining edges.
+If any of the remaining edges have `clusters.prometheus` entries, the update service will exclude the edge if the [`id` query parameter][cincinnati-for-openshift-request] is a member of the query's resulting ID set.
+To perform the PromQL request, the update service will be extended with configuration for the Prometheus connection, including a URI and authentication credentials.
+
+To reliably and efficiently return cluster-specific responses, the ID set returned by configured queries may be cached.
+There are a few possibilities for caching; although selecting a particular caching implementation is up to the update service's maintainers.
+The policy engine should refresh the cache on cache misses, as discussed in [the schema section](#enhanced-graph-data-schema-for-blocking-edges).
+
+For every graph-builder response, the policy engine may aggregate queries for all edges and warm the cache of new queries.
+This will make the initial client request faster, at the expense of maintaining a cache that may not be needed before it goes stale.
+
+The policy engine should also be prepared for, and alert on, misconfigured PromQL that results in request failures.
+In the event of such a failure, the fallback behavior should be blocking the relevant edge for all clusters until the misconfiguration is fixed.
+
+### User Stories
+
+#### Bugs which impact a subset of clusters
+
+As described in [the *Motivation* section](#motivation), enabling things like "we'd like to block this edge for clusters born in 4.1 with the `None` platform".
+
+#### OpenShift Dedicated
+
+[OpenShift Dedicated][dedicated] (OSD) restricts the recommended update sets for managed clusters more aggressively than the default service.
+For example, OSD blocked edges into 4.3.19 temporarily while [rhbz#1838007][rhbz-1838007] was investigated, while the default service did not.
+With this enhancement, OSD-specific edge blocking could be accomplished with entries like:
+
+```yaml
+to: 4.3.19
+from: .*
+clusters:
+  # show unmanaged clusters the edge, but block it from managed clusters
+  prometheus: |
+    max by (_id) (0*subscription_labels{managed="true"})
+    or
+    max by (_id) (subscription_labels{managed="false"})
+```
+
+The query returns `{_id="..."}=0` results for `managed="true"` clusters (blocking the edge) and `{_id="..."}=1` results for `managed="false"` clusters (allowing the edge).
+Clusters with both `managed="true"` and `managed="false"` records in the current time window will fall into the safer "block" bucket (the `or` semantics are defined [here][PromQL-or]).
```

Comment:
` Clusters with both managed="true" and managed="false" records in the current time window will fall into the safer "block" bucket (the or semantics are defined here).` is confusing to me. Not sue what is the implication of the statement. Specially the placement of `records in the current time window will fall into the safer "block" bucket `.

Reply:
We might have a query, which returns different results for particular cluster over the selected period (i.e. cluster has unmanaged registry and then got it installed, so any query over, say, 7d would return both "has unmanaged registry" and "has managed registry"). In order to resolve this confusion OSUS would act defensively and block the cluster until query result for a time period stabilizes

Reply:
The concern is that a cluster moves between `managed` states, and Prometheus includes both the `{_id="123...",managed="true"}` record and `{_id="123...",managed="false"}` record.  I want the query to make that a blocked cluster.  Changing state is not likely in the OSD `managed` case, but seems theoretically possible.  And it's useful to consider it anyway, because folks might pattern-match this query for other situations where clusters transitioning between allowed/blocked sets are more likely.  Added `because a cluster transitioned into or out of the managed state` with 2037152 -> b78cced.